app/app/calculations/punching.py

```diff
@@ -0,0 +1,220 @@
+    # GEOMETRIA
+    # Element = 'plyta'    # do wyboru: plyta, stopa fundamentowa TODO
+    support = support      # Bez sensu, ale poki co dla porzadku #'Słup wewnetrzny'	# Slup wewnetrzny, krawedziowy X / Y, narozny TODO  # cbColPos.index TODO
+    section = sect         # jw. # 'prostokatny'	# Przekroj prostokatny, kolowy TODO
+    cx = float(b) * cm     # cx = 40 * cm    # TODO
+    cy = float(h) * cm     #cy = 40 * cm    # TODO
+    #D = 40 * cm	       # dla przekroju kolowego TODO
+    dsw = dsw.value * mm   # srednica zbrojenia na scinanie
+    dx = float(dx) * cm
+    dy = float(dy) * cm
+    lx = float(lx) * cm    # odleglosc pola obciazenia do krawedzi plyty w kier. x
+    ly = float(ly) * cm    # jw. w kier. y
+    ad = float(ad) * cm
+    asx = float(asx) * cm2
+    asy = float(asy) * cm2    
+    lambda_u = 0
+    lambda_u = lambda_u     # wsp. uwzgledniajacy otwor w poblizu obwodu kontrolnego (%)  # TODO: to przypisanie jest tu bez sensu, ale zrobiłam tak, żeby był ślad po lambda_u 
+    alfa = pi / 2 # kat miedzy zbrojeniem na scinanie i plaszczyzna plyty (zmienic na deg!!!!!!!!!!)    # TODO
+
+    # -----------------------------------------
+    # WARUNKI PRACY
+    # TODO wszystko poniżej: musimy trzymać się konwencji pythonowych, nazwy zmiennych małą literą z podkreślnikami, jeśli chcemy coś oddzielić
+    design_sit = dsit
+    # Faktory
+    if design_sit == 'trwała' or design_sit == 'przejściowa':
+        gamma_c = 1.4
+        gamma_s = 1.15
+        alfa_cc = 1.0
+    else:   # wyjątkowa
+        gamma_c = 1.2
+        gamma_s = 1.0
+        alfa_cc = 1.0
+
+    # Obciazenia
+    beta = beta		# gdzie jest beta?
+    ved = 600 * kN	# dodac komentarz, ze nie jest to reakcja, tylko sila tnaca z plyty na slup
+                    # dla fundamentow mozna zredukowac sile do VEdred - por. 6.4.4(2)
+    # med = 50 * kNm	# TODO do rozwiniecia...
+
+
+    # ############## OBLICZENIA ##############
+    # -----------------------------------------
+    # MATERIAL
+    fcd = alfa_cc * fck / gamma_c
+
+    fyd = fyk / gamma_s
+    fywd = alfa_sw * fyk
+
+
+    # -----------------------------------------
+    # GEOMETRIA
+    d = (dx + dy) / 2 - ad	# spr. czy sie nie zgrywa z 'D' (srednica kola)
+
+    # -> u0
+    def u00(support):
+        if support == 'słup wewnetrzny':
+        # zrobic to na indeksach obiektow (1, 2, 3), zeby nie wpisywac w nieskonczonosc: wew., kraw. etc. (ryzyko pomylki)
+            return 2 * (cx + cy)
+        elif support == 'słup krawędziowy':   # Poprawic, poki co jest roboczo !!!
+            return min(2 * cx + cy, cy + 3 * d)
+        #elif support == 'słup krawędziowy X':
+        #    return min(2 * cx + cy, cy + 3 * d)
+        #elif support == 'słup krawędziowy Y':
+        #    return min(cx + 2 * cy, cx + 3 * d) 
+        elif support == 'słup narożny':
+            return min(cx + cy, 3 * d)
+        else:
+            errors.append('Blad w okresleniu polozenia slupa.')
```

Comment:
Aha, no i w tym miejscu coś trzeba zwróćić, zero czy co, trzeba się nad tym zastanowić, bo na razie jak jest błąd, to zwraca None, który później chce być przez coś mnżony i nie może :(